### PR TITLE
Make additive changes for v0.1.1

### PR DESCRIPTION
**Pull request by bhancock8**
The primary additive updates made in v0.1.1 are:

- Adding dependencies to pip setup
- Moving contrib-specific dependencies to contrib/
- Incorporation of l1 and l2 loss in Classifier._train() method
- Introducing a SparseLogisticRegression baseline
- Introducing of a RelationMention and corresponding n-gram featurizer to contrib/
- Adding optional init_kwargs and train_kwargs to ModelTuner.search()

There is also one minor non-additive update:
- Dropping the InputModule; input/middle/head modules now only need to inherit from torch.nn.Module. This should be a very minor change for anyone who has already created custom modules.

### CODE REVIEW

**Comment by ajratner** [comment]
@bhancock8 lmk when needs review!
